### PR TITLE
feat(asset): added sample codes for updating a saved query

### PR DESCRIPTION
**Pull request by weilin1995**
Added sample codes for updating a saved query

### CODE REVIEW

**Comment by telpirion** [comment]
Global feedback:
+ add correct headers

File: asset/quickstart/update-saved-query/main.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:1]: Change date to 2022.

File: asset/quickstart/update-saved-query/main.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"strconv"
+ 
+	asset "cloud.google.com/go/asset/apiv1"
+	"cloud.google.com/go/asset/apiv1/assetpb"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	field_mask "google.golang.org/genproto/protobuf/field_mask"
+)
+ 
+// Command-line flags.
+var (
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:35]: Do not use command-line set up or flags:
https://googlecloudplatform.github.io/samples-style-guide/#no-cli

File: asset/quickstart/update-saved-query/main.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	savedQueryID = flag.String("saved_query_id", "YOUR-QUERY-ID", "Identifier of Saved Query.")
+	newDescription = flag.String("new_description", "NEW-QUERY-DESCRIPTION", "New description of Saved Query.")
+)
+ 
+func main() {
+	flag.Parse()
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:41]: Remove.

File: asset/quickstart/update-saved-query/main.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	ctx := context.Background()
+	client, err := asset.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("asset.NewClient: %v", err)
+	}
+	defer client.Close()
+ 
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:49]: Don't use env vars inside of a sample. We use them for testing, true; but we don't expect customers to set anything other than ADC.

File: asset/quickstart/update-saved-query/main.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	cloudresourcemanagerClient, err := cloudresourcemanager.NewService(ctx)
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:50]: Needs a `defer cloudResourceManagerClient.close()` after the err check.

Comment (telpirion) [reply asset/quickstart/update-saved-query/main.go:50]: Be sure to use camelCase for variables, esp. product names.

File: asset/quickstart/update-saved-query/main.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	if err != nil {
+		log.Fatalf("cloudresourcemanager.NewService: %v", err)
+	}
+ 
+	project, err := cloudresourcemanagerClient.Projects.Get(projectID).Do()
+	if err != nil {
+		log.Fatalf("cloudresourcemanagerClient.Projects.Get.Do: %v", err)
+	}
+	projectNumber := strconv.FormatInt(project.ProjectNumber, 10)
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main.go:59]: Why do we need to get the project number? Can't we just use the project ID? I'd add a comment here explaining why this is needed.

File: asset/quickstart/update-saved-query/main_test.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	"testing"
+	"time"
+ 
+	asset "cloud.google.com/go/asset/apiv1"
+	"cloud.google.com/go/asset/apiv1/assetpb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+)
+ 
+func TestMain(t *testing.T) {
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main_test.go:31]: `TestMain()` in go is used for set up and teardown. In this case, the test case should have the format: `TestUpdateSavedQuery(t *testing.T).

File: asset/quickstart/update-saved-query/main_test.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+	}
+ 
+	m := testutil.BuildMain(t)
+	defer m.Cleanup()
+ 
+	if !m.Built() {
+		t.Errorf("failed to build app")
+	}
+	newDescription := "This is a new description"
+	stdOut, stdErr, err := m.Run(env, 2*time.Minute, fmt.Sprintf("--saved_query_id=%s --new_description=%s", queryId, newDescription))
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main_test.go:82]: Remove this CLI invocation and call the sample method directly.

**Review by telpirion — CHANGES_REQUESTED** [review]
Please take a look at the following before re-requesting a review:

+ [canonical Go sample](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/dlp/snippets/inspect/inspect_string.go)
+ [All-Cloud style guide](https://googlecloudplatform.github.io/samples-style-guide)
+ [Effective Go](https://go.dev/doc/effective_go)

File: asset/quickstart/update-saved-query/main_test.go [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+	buf := new(bytes.Buffer)
+	newDescription := "This is a new description"
+	update_err := updateSavedQuery(buf, projectID, savedQueryID, newDescription)
+
+	fullQueryName := fmt.Sprintf("projects/%s/savedQueries/%s", projectNumber, savedQueryID)
+	delete_err := client.DeleteSavedQuery(ctx, &assetpb.DeleteSavedQueryRequest{
+		Name: fullQueryName,
+	})
+	if update_err != nil {
+		t.Errorf("updateSavedQuery failed: %v", update_err)
```
Comment (telpirion) [inline asset/quickstart/update-saved-query/main_test.go:98]: NIT: Change this to `t.Fatalf()`. If you error here, everything else will presumably fail, too.

**Review by telpirion — APPROVED** [review]

File: asset/quickstart/update-saved-query/main.go (hunk 1 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: changed.

File: asset/quickstart/update-saved-query/main.go (hunk 2 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: changed.

File: asset/quickstart/update-saved-query/main.go (hunk 3 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: removed.

File: asset/quickstart/update-saved-query/main.go (hunk 4 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: changed in the latest revision:)

File: asset/quickstart/update-saved-query/main.go (hunk 5 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: changed in the latest revision:)

File: asset/quickstart/update-saved-query/main.go (hunk 6 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main.go]: added comments:)

File: asset/quickstart/update-saved-query/main_test.go (hunk 7 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main_test.go]: changed in the latest revision:)

File: asset/quickstart/update-saved-query/main_test.go (hunk 8 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main_test.go]: changed in the latest revision:)

File: asset/quickstart/update-saved-query/main_test.go (hunk 9 above)
Comment (weilin1995) [reply asset/quickstart/update-saved-query/main_test.go]: changed in the latest revision:)

**Pull request merged**